rtgui/gradient.cc

```diff
@@ -136,88 +136,99 @@ void Gradient::read (const ProcParams* pp, const ParamsEdited* pedited)
-        currLine = static_cast<Line*>(mouseOverGeometry.at(0));
-        polCoord1.set(1500.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1);
-        currLine->begin += origin;
-        polCoord1.set(1500.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1);
-        currLine->end   += origin;
-        // update vertical line
-        currLine = static_cast<Line*>(visibleGeometry.at(1));
-        polCoord1.set( 700.f, float(-degree_ + 90 ));
-        currLine->begin.setFromPolar(polCoord1);
-        currLine->begin += origin;
-        polCoord1.set( 700.f, float(-degree_ + 270));
-        currLine->end.setFromPolar  (polCoord1);
-        currLine->end   += origin;
-        currLine = static_cast<Line*>(mouseOverGeometry.at(1));
-        polCoord1.set( 700.f, float(-degree_ + 90 ));
-        currLine->begin.setFromPolar(polCoord1);
-        currLine->begin += origin;
-        polCoord1.set( 700.f, float(-degree_ + 270));
-        currLine->end.setFromPolar  (polCoord1);
-        currLine->end   += origin;
-        // update upper feather line
-        currLine = static_cast<Line*>(visibleGeometry.at(2));
-        polCoord2.set(decay, float(-degree_ + 270));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        currLine = static_cast<Line*>(mouseOverGeometry.at(2));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        // update lower feather line
-        currLine = static_cast<Line*>(visibleGeometry.at(3));
-        polCoord2.set(decay, float(-degree_ + 90));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        currLine = static_cast<Line*>(mouseOverGeometry.at(3));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        // update circle's position
-        currCircle = static_cast<Circle*>(visibleGeometry.at(4));
-        currCircle->center = origin;
-        currCircle = static_cast<Circle*>(mouseOverGeometry.at(4));
-        currCircle->center = origin;
+        if (!imW || !imH) {
```

Comment:
I know `ImageArea::getImageSize` nulls them, but I would feel better if `imW` and `imH` were properly initialized to zero, especially since `getImageSize` is a virtual method and might collect further overrides.